### PR TITLE
Adding a back to top button feature

### PR DESCRIPTION
**Pull request by afzalsayed96**
## Brief description

Due to length of index.html a back-to-top button was long needed. This feature is fully implemented using CSS only. It uses hover to control visibility of the button.

## Developer Certificate of Origin

- [x] I certify that these changes according to the Developer Certificate of Origin 1.1 as described at <https://developercertificate.org/>.

## Sample pictures

demo

## Further details

Had to add ~a `back2top.less` file and~ a class `.to-top` to control the behavior of button

### CODE REVIEW

**Comment by afzalsayed96** [comment]
mixins?

File: src/back2top.less [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    opacity: 1;
+    display: inline-block;
+    padding: 1em;
+    position: fixed;
+    bottom: 1em;
+    right: 1em;
+    .paper-btn {
+    padding: 0.6em 1em;
+    background: white;
+    border-top-left-radius: 185px 160px;
```
Comment (Fraham) [inline src/back2top.less]: As @TotomInc  was saying i think this should use the .border class so it makes the same style across the framework

File: src/back2top.less [hunk 2]
```diff
@@ -0,0 +1,20 @@
+.to-top {
+    opacity: 1;
+    display: inline-block;
+    padding: 1em;
+    position: fixed;
+    bottom: 1em;
+    right: 1em;
+    .paper-btn {
+    padding: 0.6em 1em;
```
Comment (Fraham) [inline src/back2top.less]: could these lines be indented to show that its within the paper-btn class

File: src/back2top.less [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    background: white;
```
Comment (Fraham) [inline src/back2top.less]: Could this use the variable instead of hardcoded value

Comment (afzalsayed96) [reply src/back2top.less]: variable for color after defining in colors.less, right?

File: src/back2top.less (hunk 1 above)
Comment (afzalsayed96) [reply src/back2top.less]: I didn't use .border in order to make the button a circle. Adding .border would make it like a square (which is consistent across framework)
So should I
1. Make the button square - using `.border' class?
2. Define 'circle-border' in borders.less and keep using circle button?

File: src/back2top.less (hunk 3 above)
Comment (afzalsayed96) [reply src/back2top.less]: White color is not defined in colors.less. Also this is used in container.less 
```
.paper {
background: #FFF
}
```

**Comment by rhyneav** [comment]
Very nice and very needed while this is still a monster index.html file. Since this is unique to the documentation page, would it be best to move this into the demo.css? Or could this be a feature that can be used by others in their sites?

**Comment by afzalsayed96** [comment]
I'm confused too about whether this could be a feature or not.
BTW what does demo.css do?

**Comment by TotomInc** [comment]
@afzalsayed96 `demo.css` is an independant file from the framework, which is only used in the docs/demo app for demonstration purposes.

**Comment by afzalsayed96** [comment]
@tomtominc thanks for clarification :)

File: src/back2top.less (hunk 2 above)
Comment (afzalsayed96) [reply src/back2top.less]: Done

File: src/back2top.less (hunk 3 above)
Comment (afzalsayed96) [reply src/back2top.less]: Done

**Comment by rhyneav** [comment]
Awesome, thank you!

**Pull request merged**